### ISSUE TITLE
Material 3 focus rings & keyboard behavior on web/desktop

### ISSUE THREAD

**Issue by guidezpl** [flutter/flutter#120425]
This is a placeholder issue for a project to draw outlined indicators ("focus rings") around focused M3 elements. These improve a11y and ease of use.

For example, with `NavigationDrawer`, it would look as follows
![image](https://user-images.githubusercontent.com/6655696/218030566-03e49648-4bc5-4ee6-9ad6-8efc34114b4c.png)

There are also slight refinements to keyboard shortcuts. For example, with `NavigationDrawer`, the following is expected:

Keys | Actions
-- | --
Tab | Focus lands on the first navigation destination
Space | Selects the focused navigation destination
Enter | Navigate between destinations within the navigation drawer

For more, Googlers see go/m3-web-focus-rings and go/focus-outlines-material-android-views.

**Comment by rodydavis** [comment]
Just wanted to also bring up the focus ring on the web and how it might apply here.

A focus ring is used any time a keyboard is used or voice over is applied.

The focus ring disappears if the screen it touched.

**Example**: iPad with keyboard or windows touch screen laptop (focus ring on TAB, disappear on touch)

Here is a codepen with M3 and CSS for a focus ring:
https://codepen.io/rodydavis/pen/BaxPvbq

About the outlines themselves, it is common to want to hide the outline when accessible mode is not active by setting the outline color to transparent. But when the user has a high contrast mode applied the transparent color will be set to the default color.

In addition the web supports :focus-within which will highlight the top level component that has focus, not the current child. Not sure if that is how it currently works with Flutter.
https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-within

**Comment by HansMuller** [comment]
CC @gspencergoog

**Comment by gspencergoog** [comment]
Flutter currently hides focus indicators when the FocusManager.highlightMode is FocusHighlightMode.touch, which happens whenever the device is touched (i.e. a pointer event with a pointer kind of "touch", "stylus", or "invertedStylus" is received).

Buttons, text fields, and other widgets that show focus indicators listen for changes in the highlight mode.

**Comment by slightfoot** [comment]
Just did an example of this on Humpday. https://gist.github.com/slightfoot/907156540cf1e4ec60dbd96104769098

**Comment by Feichtmeier** [comment]
@Piinks even though the title includes "web" this is highly requested by the canonical desktop design team and our yaru.dart theme and widget package. Could you eventually add the desktop tag? Thanks in advance

**Comment by paul-geoghegan** [comment]
Hi there, I'm wondering if we could hear more about the progress for this as @juanruitina mentioned the current solution does not meet contrast guidelines and I don't think there is an easy way to disable the default focus to add your own custom focus rings without the padding staying in place etc.
